src/ORM/RulesChecker.php

```diff
@@ -89,4 +90,27 @@ public function existsIn($field, $table, $message = null)
         $errorField = is_string($field) ? $field : current($field);
         return $this->_addError(new ExistsIn($field, $table), '_existsIn', compact('errorField', 'message'));
     }
+
+    /**
+     * Validates the count of associated records.
+     *
+     * @param The entity property that contains the associations.
+     * @param int The expected count.
+     * @param string The operator for the count comparison.
+     * @param string|null $message The error message to show in case the rule does not pass.
+     * @return callable
+     */
+    public function validCount($field, $count = 0, $operator = '>', $message = null)
+    {
+        if (!$message) {
+            if ($this->_useI18n) {
+                $message = __d('cake', 'The count does not match {0}{1}', [$operator, $count]);
+            } else {
+                $message = 'This value does not exist';
```

Comment:
> But then you're not telling the user what exactly is wrong.

In that case shouldn't the untranslated message be similar to the translated one?